app/src/userpages/components/Avatar/NameAndUsername/index.jsx

```diff
@@ -23,9 +24,11 @@ const NameAndUsername = ({ name, username, children }: NameAndUsernameProps) =>
             <div className={styles.usernameWrapper}>
                 <span title={username} className={styles.username}>
                     {!isEthAddress && (username)}
-                    {!!isEthAddress && (truncate(username, {
-                        maxLength: 20,
-                    }))}
+                    <HoverCopy value={username}>
```

Comment:
This always generates a `<div>` which makes the structure dirty if we're rocking username only. I'd consider doing
```jsx
{isEthAddress ? (
    <HoverCopy …>
        {/* Here we can go with unconditioned `truncate` */}
        {truncate(username, …)}
    </HoverCopy>
) : (
    username
)}
```

^^ makes it _either-or_.